src/cose/nss.rs

```diff
@@ -277,14 +278,19 @@ pub fn sign(
     let hash_buf = hash(payload).unwrap();
     let hash_item = SECItem::maybe_new(hash_buf.as_slice())?;
     let signature_len = unsafe { PK11_SignatureLen(key) };
-    let mut signature: Vec<u8> = Vec::with_capacity(signature_len);
-    // XXX: rewrite without unsafe.
-    unsafe {
-        signature.set_len(signature_len);
-    }
+    // We indirectly pass this to PK11_SignWithMechanism and it gets modified,
+    // so this has to be mut.
+#[allow(unused_mut)]
```

Comment:
Hm, I expected this to work. But I still get the warning. It works when I put it on the function (but that's probably not what we want).

Reply:
Yeah, it turns out this wasn't the right approach anyway. I'll push an update shortly.